action.yaml

```diff
@@ -95,20 +100,59 @@ runs:
+        INPUT_GITHUB_TOKEN=$INPUT_GITHUB_TOKEN
+        INPUT_TRUNK_TOKEN=$INPUT_TRUNK_TOKEN
+        EOF
+
+        # Every inputs.field should be referenced as INPUT_FIELD later in the action. This allows
+        # the field to be set either as an argument to the github action or via inputs.json.
+        if [[ -n "${{ fromJSON(inputs.json).version }}" ]]; then
+
+          cat >>$GITHUB_ENV <<EOF
+        GITHUB_EVENT_PULL_REQUEST_BASE_REPO_OWNER=${{ fromJSON(inputs.json).pull_request.base.repo.owner.login }}
+        GITHUB_EVENT_PULL_REQUEST_BASE_REPO_NAME=${{ fromJSON(inputs.json).pull_request.base.repo.name }}
+        GITHUB_EVENT_PULL_REQUEST_BASE_SHA=${{ fromJSON(inputs.json).pull_request.base.sha }}
+        GITHUB_EVENT_PULL_REQUEST_HEAD_REPO_FORK=${{ fromJSON(inputs.json).pull_request.head.repo.fork }}
+        GITHUB_EVENT_PULL_REQUEST_HEAD_SHA=${{ fromJSON(inputs.json).pull_request.head.sha }}
+        GITHUB_EVENT_PULL_REQUEST_NUMBER=${{ fromJSON(inputs.json).pull_request.number }}
+        INPUT_ARGUMENTS=${{ fromJSON(inputs.json).arguments }}
+        INPUT_CACHE=${{ fromJSON(inputs.json).cache }}
+        INPUT_CHECK_ALL_MODE=${{ fromJSON(inputs.json).check_all_mode }}
+        INPUT_CHECK_MODE=${{ fromJSON(inputs.json).check_mode }}
+        INPUT_CHECK_RUN_ID=${{ fromJSON(inputs.json).check_run_id }}
+        INPUT_GITHUB_REF_NAME=${{ fromJSON(inputs.json).target.ref_name }}
+        INPUT_LABEL=${{ fromJSON(inputs.json).label }}
+        INPUT_TRUNK_PATH=${{ fromJSON(inputs.json).trunk_path }}
+        INPUT_UPLOAD_SERIES=${{ fromJSON(inputs.json).upload_series }}
+        EOF
+
+        else
+
+          cat >>$GITHUB_ENV <<EOF
+        GITHUB_EVENT_PULL_REQUEST_BASE_SHA=${{ github.event.pull_request.base.sha }}
+        GITHUB_EVENT_PULL_REQUEST_HEAD_REPO_FORK=${{ github.event.pull_request.head.repo.fork }}
+        GITHUB_EVENT_PULL_REQUEST_HEAD_SHA=${{ github.event.pull_request.head.sha }}
+        GITHUB_EVENT_PULL_REQUEST_NUMBER=${{ github.event.pull_request.number }}
+        GITHUB_REF_NAME=${{ github.ref_name }}
         INPUT_ARGUMENTS=${{ inputs.arguments }}
-        INPUT_GITHUB_REF_NAME=${{ env.INPUT_GITHUB_REF_NAME || github.ref_name }}
+        INPUT_CACHE=${{ inputs.cache }}
+        INPUT_CHECK_ALL_MODE=${{ inputs.check-all-mode }}
+        INPUT_CHECK_MODE=${{ inputs.check-mode }}
+        INPUT_CHECK_RUN_ID=${{ inputs.check-run-id }}
+        INPUT_GITHUB_REF_NAME=${{ github.ref_name }}
         INPUT_LABEL=${{ inputs.label }}
-        INPUT_TRUNK_TOKEN=$INPUT_TRUNK_TOKEN
+        INPUT_TRUNK_PATH=${{ inputs.trunk-path }}
+        INPUT_UPLOAD_SERIES=${{ inputs.upload_series }}
         EOF
+
+        fi
       env:
-        INPUT_GITHUB_TOKEN: ${{ inputs.github-token }}
-        INPUT_TRUNK_TOKEN: ${{ inputs.trunk-token }}
+        INPUT_GITHUB_TOKEN: ${{ inputs.github-token || fromJSON(inputs.json).github_token }}
+        INPUT_TRUNK_TOKEN: ${{ inputs.trunk-token || fromJSON(inputs.json).token }}
```

Comment:
Shouldn't these two be up above?

Reply:
added explanatory comment - we have to add-mask these, so these are handled differently